apps/src/code-studio/videos.js

```diff
@@ -274,6 +293,33 @@ videos.showVideoDialog = function(options, forceShowVideo) {
   });
 };
 
+/**
+ * Add a URL parameter (or changing it if it already exists)
+ * @param {search} string  this is typically document.location.search
+ * @param {key}    string  the key to set
+ * @param {val}    string  value
+ */
+var addUrlParam = function(search, key, val) {
+  var newParam = key + '=' + val,
+    params = '?' + newParam;
+
+  // If the "search" string exists, then build params from it
+  if (search) {
+    // Try to replace an existance instance
+    params = search.replace(
+      new RegExp('([?&])' + key + '[^&]*'),
+      '$1' + newParam
+    );
```

Comment:
Nonblocking for this use case: This function fails in the case where one queryparam key is a subset of another; trying to replace `part1=foo` might replace `part10=bar` instead if it showed up first in the search string.